zeebe/backup-stores/s3/src/main/java/io/camunda/zeebe/backup/s3/S3BackupStore.java

```diff
@@ -387,29 +383,34 @@ private CompletableFuture<FileSet> saveSegmentFiles(final Backup backup) {
   }
 
   public static S3AsyncClient buildClient(final S3BackupConfig config) {
-    final var builder = S3AsyncClient.builder();
+    final var builder = S3AsyncClient.crtBuilder();
 
-    // Enable auto-tuning of various parameters based on the environment
-    builder.defaultsMode(DefaultsMode.AUTO);
+    // Note: The CRT-based client has a different configuration API than the standard client.
+    // The CRT client automatically optimizes for large file transfers.
 
-    // Enable legacy MD5 plugin if configured, as it is required for compatibility with S3
-    // compatible storage systems that expect MD5 checksums in the request.
-    if (config.supportLegacyMd5()) {
```

Comment:
:x: you cannot remove md5 support as it's a mandatory feature, see the [test runs](https://github.com/camunda/camunda/actions/runs/18973765082/job/54189258396?pr=40319)